### PR TITLE
Multiple fixes

### PR DESCRIPTION
**Pull request by leojrfs**
All changes are currently tested on macOS 10.13.1, and different distros of linux, x86_64 and ARM, using Contour Next USB and Contour Next One.

### CODE REVIEW

File: src/hid_commands.cpp [hunk 1]
```diff
@@ -328,6 +330,12 @@ bool interface::sync(const char*& result_begin, const char*& result_end)
     if (parsed) { // parsed frame, send ACK
       write(hid_, ACK);
       state_ = data;
+      if (result_begin != NULL) { // Message Terminator Record frame received, write CAN, done
```
Comment (drougas) [inline src/hid_commands.cpp:333]: For some reason, my Contour Next USB returns "Software Error: Turn off then on. If error persists call Customer Service". The error is E86. Let me check it some more (couple of days?)...

File: cmake/Modules/FindHidapi.cmake [hunk 2]
```diff
@@ -1,57 +1,30 @@
...
-	ENDIF( NOT HIDAPI_FIND_QUIETLY )
-ELSE(HIDAPI_FOUND)
-	IF( HIDAPI_FIND_REQUIRED)
-		MESSAGE( FATAL_ERROR "Could not find Hidapi" )
-	ELSE( HIDAPI_FIND_REQUIRED)
-		MESSAGE( STATUS "Optional package Hidapi was not found" )
-	ENDIF( HIDAPI_FIND_REQUIRED)
-ENDIF(HIDAPI_FOUND)
-
+message (STATUS "Checking for HIDAPI...")
```
Comment (drougas) [inline cmake/Modules/FindHidapi.cmake:1]: Thanks, this was really needed!

**Review by drougas — COMMENTED** [review]
Need to check for E86 in the case of Contour Next USB. It's not that important (it appears after sync finishes successfully), but let me check it a bit more.

File: src/hid_commands.cpp (hunk 1 above)
Comment (leojrfs) [reply src/hid_commands.cpp:333]: Hmm, maybe its because of the CAN at the end. Can you share the debug output?

Comment (drougas) [reply src/hid_commands.cpp:333]: Indeed, it's the CAN at the end. I removed it and it works fine with the Contour Next. I'll check-in your changes and open a new issue to fix that.

**Review by drougas — APPROVED** [review]

**Pull request merged**